imports/ui/case/case-master.jsx

```diff
@@ -82,8 +78,22 @@ class CaseMaster extends Component {
   handleIconClick = () => {
     this.props.dispatch(setDrawerState(true))
   }
+
+  handleSearch = (searchText) => {
+    this.setState({ searchText: searchText })
+    if (searchText === '') {
+      this.setState({ searchMode: false })
+    } else {
+      this.setState({ searchMode: true })
+      const matcher = new RegExp(searchText, 'i')
+      const searchResult = this.state.componentsProps['/case'].caseList
```

Comment:
Please change it so only the search term (whatever the user typed) is sent and shared via Redux.

The way it is currently done is more convoluted and mixing up unrelated concerns in the less relevant places:
* The "CaseMaster" gets intimate knowledge of the internal data structure its components are using (consuming "caseList" by the mechanism I created for sharing data between components and different versions of their header bar).
* Search/filtering logic is mis-located, as it should be done where the data is being prepared to be rendered (you need to move it into CaseExplorer)